Formula/llvm.rb

```diff
@@ -108,33 +128,54 @@ class Llvm < Formula
   keg_only :provided_by_osx
 
   option :universal
-  option "with-clang", "Build the Clang compiler and support libraries"
-  option "with-clang-extra-tools", "Build extra tools for Clang"
-  option "with-compiler-rt", "Build Clang runtime support libraries for code sanitizers, builtins, and profiling"
-  option "with-libcxx", "Build the libc++ standard library"
-  option "with-lld", "Build LLD linker"
+  option "without-clang", "Do not build the Clang compiler and support libraries"
+  option "without-clang-extra-tools", "Do not build extra tools for Clang"
+  option "without-compiler-rt", "Do not build Clang runtime support libraries for code sanitizers, builtins, and profiling"
+  option "without-libcxx", "Do not build the libc++ standard library"
+  option "with-libcxxabi", "Build the libc++abi standard library"
+  option "with-libunwind", "Build the libunwind library"
+  option "without-lld", "Do not build LLD linker"
   option "with-lldb", "Build LLDB debugger"
+  option "without-openmp", "Build without the additional OpenMP Runtime Libraries"
   option "with-python", "Build Python bindings against Homebrew Python"
-  option "with-rtti", "Build with C++ RTTI"
-  option "with-utils", "Install utility binaries"
-
-  deprecated_option "rtti" => "with-rtti"
-
-  if MacOS.version <= :snow_leopard
+  option "without-rtti", "Build without C++ RTTI"
+  option "without-utils", "Do not install utility binaries"
+  option "without-polly", "Build without Polly optimizer"
+  option "with-test", "Build LLVM unit tests"
+  option "with-shared-libs", "Build shared instead of static libraries"
+  option "without-libffi", "Use libffi to call external functions from the interpreter"
+  option "with-all-targets", "Build all targets rather than just AMDGPU, ARM, NVPTX, and X86"
+
+  depends_on "libffi" => :recommended # llvm.org/docs/GettingStarted.grml#requirements
+  depends_on "graphviz" => :optional # for the 'dot' tool (lldb)
+  depends_on "ocaml" => :optional
+
+  if build.with? "python"
+    depends_on "python"
```

Comment:
Let's make it simple:
`build.with? "python"` means building against python in path, i.e. current `python-config` approach.
`build.without? "python"` means building against system python, i.e. do not set any python env as the same behavior as original llvm.rb.

Basically
```
if build.with? "lldb"
 if build.with? "python"
   args << "-DLLDB_RELOCATABLE_PYTHON=ON"
   args << "-DPYTHON_LIBRARY=#{pylib}"
   args << "-DPYTHON_INCLUDE_DIR=#{pyinclude}"
  end
end
```